proxy_agent/src/host_clients/wire_server_client.rs

```diff
@@ -56,7 +56,9 @@ impl WireServerClient {
         }
 
         let url = format!("http://{}:{}/{}", self.ip, self.port, TELEMETRY_DATA_URI);
-        let url: Uri = url.parse().map_err(|e| Error::parse_url(url, e))?;
+        let url: Uri = url
+            .parse()
+            .map_err(|e: InvalidUri| Error::ParseUrl(url, e.to_string()))?;
```

Comment:
I see you explicitly add the error type for InvalidUri in many places. is it on purpose and required to make code built?

Reply:
Please update them like 
`        let url = url
            .parse::<Uri>()
            .map_err(|e| Error::ParseUrl(url, e.to_string()))?;`